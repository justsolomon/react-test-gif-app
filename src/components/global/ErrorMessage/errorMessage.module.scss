@import "../../../assets/scss/base/variables";

.error-message {
  text-align: center;

  svg,
  &__description {
    margin-bottom: 16px;
  }

  svg {
    height: 28px;
    width: 28px;
    color: $tertiary-color;
  }

  &__description {
    line-height: 20px;
  }

  &__button {
    border-radius: 20px;
  }
}
