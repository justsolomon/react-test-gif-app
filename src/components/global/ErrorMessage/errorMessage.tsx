import { Button } from "../Button";
import styles from "./errorMessage.module.scss";
import { ReactComponent as ErrorIcon } from "assets/vectors/error.svg";

interface IErrorMessage {
  retryRequest: () => void;
  error: string;
}

function ErrorMessage({ retryRequest, error }: IErrorMessage) {
  return (
    <div className={styles["error-message"]}>
      <ErrorIcon />
      <p className={styles["error-message__description"]}>
        {error === "Network Error"
          ? "Looks like you lost your connection. Please check it and try again"
          : error}
      </p>
      <Button className="error-message__button" onClick={retryRequest}>
        Try again
      </Button>
    </div>
  );
}

export default ErrorMessage;
