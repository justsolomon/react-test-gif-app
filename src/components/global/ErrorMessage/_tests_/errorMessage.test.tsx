import { render } from "@testing-library/react";
import ErrorMessage from "../errorMessage";

describe("<ErrorMessage />", () => {
  it("should render the error message component properly", () => {
    const { asFragment } = render(
      <ErrorMessage error="An error occurred" retryRequest={() => {}} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
