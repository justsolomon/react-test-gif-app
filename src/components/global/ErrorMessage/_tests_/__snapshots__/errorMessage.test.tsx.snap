// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should render the error message component properly 1`] = `
<DocumentFragment>
  <div
    class="error-message"
  >
    <svg>
      error.svg
    </svg>
    <p
      class="error-message__description"
    >
      An error occurred
    </p>
    <button
      class="button error-message__button button--solid button--none "
      type="button"
    >
      Try again
    </button>
  </div>
</DocumentFragment>
`;
